numba/tests/test_caching.py

```diff
@@ -701,6 +704,87 @@ def execute_with_input():
         err = execute_with_input()
         self.assertIn("cache hits = 1", err.strip())
 
+    def test_zip_caching(self):
+        import importlib
+        import sys
+
+        # Create a simple Python module to be zipped
+        mod_content = """
+from numba import jit
+
+@jit(cache=True)
+def add(x, y):
+    return x + y
+"""
+        mod_filename = "test_module.py"
+        zip_filename = "test_archive.zip"
+
+        # Create a zip file containing the module
+        zip_path = os.path.join(self.tempdir, zip_filename)
+        with zipfile.ZipFile(zip_path, "w") as zf:
+            zf.writestr(mod_filename, mod_content)
+
+        # Add the zip file to sys.path
+        sys.path.insert(0, zip_path)
+
+        try:
+            # First import and call
+            import test_module  # type: ignore
+
+            result1 = test_module.add(2, 3)
+            self.assertEqual(result1, 5)
+            self.assertEqual(
+                test_module.add.stats.cache_hits[(type(None), type(None))], 0
+            )
+
+            # Record the initial cache hits
+            initial_cache_hits = dict(test_module.add._cache_hits)
+
+            # Remove the module and reimport
+            del sys.modules["test_module"]
```

Comment:
Maybe you can use `self.import_module()` here. As it will re-import the module.

Reply:
I tried but because of the checks that `self.import_module()` does, it fails for the zip. I added a note for future travellers...